apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/profile/ProfileTaskExecutionContext.java

```diff
@@ -30,22 +37,139 @@
     // task data
     private final ProfileTask task;
 
-    // task real start time
-    private final long startTime;
+    // record current profiling count, use this to check has available profile slot
+    private final AtomicInteger currentProfilingCount = new AtomicInteger(0);
+
+    // profiling segment slot
+    private volatile ThreadProfiler[] profilingSegmentSlots = new ThreadProfiler[Config.Profile.MAX_PARALLEL];
+
+    // current profiling execution future
+    private volatile Future profilingFuture;
+
+    // total started profiling tracing context count
+    private final AtomicInteger totalStartedProfilingCount = new AtomicInteger(0);
 
-    public ProfileTaskExecutionContext(ProfileTask task, long startTime) {
+    public ProfileTaskExecutionContext(ProfileTask task) {
         this.task = task;
-        this.startTime = startTime;
+    }
+
+    /**
+     * start profiling this task
+     * @param executorService
+     */
+    public void startProfiling(ExecutorService executorService) {
+        profilingFuture = executorService.submit(new ProfileThread(this));
+    }
+
+    /**
+     * stop profiling
+     */
+    public void stopProfiling() {
+        if (profilingFuture != null) {
+            profilingFuture.cancel(true);
+        }
+    }
+
+    /**
+     * check have available slot to profile and add it
+     * @param tracingContext
+     * @param firstSpanOPName
+     * @return
+     */
```

Comment:
You have so many JavaDocs like this, please add corresponding descriptions, or remove the `@param` and `@return`, although it can compile, it produces uncountable warnings in CI (`javadoc:javadoc`), and many many warnings in the IDE, making it hard to discover warnings that really matters:

<img width="634" alt="image" src="https://user-images.githubusercontent.com/15965696/72618132-6bc12b80-3975-11ea-8545-62f6220cdb96.png">